.github/workflows/npmpublish.yml

```diff
@@ -0,0 +1,75 @@
+# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
+# For more information see: https://help.github.com/actions/language-and-framework-guides/publishing-nodejs-packages
+
+name: Node.js Package
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js
+      uses: actions/setup-node@v1
+    - name: Setup xvfb
+      run: |
+        Xvfb :99 -screen 0 1024x768x24 &
+        echo "::set-env name=DISPLAY:::99.0"
+    - name: Test
+      run: |
+        npm ci
+        npm test
+
+  publish-npm:
+    needs: test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+          registry-url: https://registry.npmjs.org/
+      # Since we created the release in the UI, we need to find it.
+      # This step gets the release from the GITHUB_REF env var
+      - name: Get release
+        id: get_release
+        uses: bruceadams/get-release@v1.2.2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Setup and build
+        run: |
+          npm ci
+          npm install -g json
+          json -I -f package.json -e "this.version=\"$GITHUB_REF\""
+          json -I -f package-lock.json -e "this.version=\"$GITHUB_REF\""
+          npm run build
+          ./scripts/docs-config.sh "${GITHUB_REF:1}"
+          npm run docs
+          npm run typedoc
+          npm pack
+      - name: Publish to NPM
```

Comment:
I realize that having separate steps here might break this. https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-output-parameter implies that the machine is reset between steps but I can't figure out if that means all the files are removed or not